### ISSUE TITLE
Removing key files, and cleaning up the results directory structure.

### ISSUE THREAD

**Issue by jaredgalloway** [matsengrp/gcreplay#95]
TLDR - I'm doing some code cleanup items that I feel are necessary before tackling the other open issues. I'm getting close on this, and then the other issues should move more quickly without the extra cognitive load this has been causing.

**Description of changes:**
Time to bite the bullet and refactor the main pipeline, just a bit. Working with the rushed choices that were made early-on in the replay pipeline is no longer really feasible. Things that should be easy are simply taking too much time, are error-prone, and generally confusing for anyone else who may pick up this code in the future:

- Read from the metadata.csv file directly, rather than manually generating annoying key files that contain the same general information (but annoyingly different column specification). This solves the problem brought up [here](https://github.com/matsengrp/gcreplay/issues/89). And this way we'll just have two simple inputs to the pipeline: an NGS manifest, and GC metadata.
- Independent python scripts for each step in the pipeline rather than each sharing access to a single module. This blows the entire workflow cache (many hours of compute) each time the module is changed.
- Compression and cleaning up LFS history so old, large files don't clog the repo (we have local archives for older results). We're in this annoying place where PR's on [github.com](http://github.com/) often crash chrome because there are so many uncompressed non-binary files getting updated in the results directory.
